zmon_agent/discovery/kubernetes/cluster.py

```diff
@@ -400,3 +409,60 @@ def get_cluster_daemonsets(kube_client, cluster_id, alias, region, infrastructur
         entities.append(entity)
 
     return entities
+
+
+def list_postgres_databases(*args, **kwargs):
+    conn = psycopg2.connect(*args, **kwargs)
+    cur = conn.cursor()
+    cur.execute("""
+        SELECT datname
+          FROM pg_database
+         WHERE datname NOT IN('postgres', 'template0', 'template1')
+    """)
+    return [row[0] for row in cur.fetchall()]
+
+
+def get_cluster_postgresdbs(kube_client, cluster_id, alias, region, infrastructure_account,
+                            postgres_user, postgres_pass,
+                            namespace=None):
+    entities = []
+
+    services = get_all(kube_client, kube_client.get_services, namespace)
+
+    for service in services:
+        obj = service.obj
+
+        # TODO: filter in the API call
+        labels = obj['metadata'].get('labels', {})
+        if labels.get('application') != 'spilo':
+            continue
+
+        service_namespace = obj['metadata']['namespace']
+        service_dns_name = '{}.{}.svc.cluster.local'.format(service.name, service_namespace)
+        service_port = obj['spec']['ports'][0]['port']  # Assume the first port is the one we need.
+
+        dbnames = list_postgres_databases(host=service_dns_name,
+                                          port=service_port,
+                                          user=postgres_user,
+                                          password=postgres_pass,
+                                          dbname='postgres')
+        for db in dbnames:
+            entity = {
+                'id': 'postgresdb-{}-{}-{}[{}]'.format(db, service.name, service.namespace, cluster_id),
+                'type': POSTGRESDB_TYPE,
+                'kube_cluster': cluster_id,
+                'alias': alias,
+                'created_by': AGENT_TYPE,
+                'infrastructure_account': infrastructure_account,
+                'region': region,
+
```

Comment:
This actually just tries to follow the style of the rest of functions in this file: the id/region/etc. common fields go first, then entity-specific ones are separated with a blank line.  E.g: https://github.com/zalando-zmon/zmon-agent-core/blob/master/zmon_agent/discovery/kubernetes/cluster.py#L203